added template for better error explanation to assert.ok error message

chrbergert opened the pull request:
<!--
Thank you for your pull request. Please provide a description above and review
the requirements below.

Bug fixes and new features should include tests and possibly benchmarks.

Contributors guide: https://github.com/nodejs/node/blob/master/CONTRIBUTING.md
-->

##### Checklist
<!-- Remove items that do not apply. For completed items, change [ ] to [x]. -->

- [ ] `make -j4 test` (UNIX), or `vcbuild test` (Windows) passes
- [ ] tests and/or benchmarks are included
- [ ] documentation is changed or added
- [ ] commit message follows [commit guidelines](https://github.com/nodejs/node/blob/master/CONTRIBUTING.md#commit-message-guidelines)

##### Affected core subsystem(s)
<!-- Provide affected core subsystem(s) (like doc, cluster, crypto, etc). -->

joyeecheung commented on test/parallel/test-child-process-spawnsync-maxbuf.js:
```diff
@@ -16,7 +16,7 @@ const args = [
   const ret = spawnSync(process.execPath, args, { maxBuffer: 1 });
 
-  assert.ok(ret.error, 'maxBuffer should error');
+  assert.ok(ret.error, `instead of an error there is ${ret.error}` );
```
Since `ret` is returned by `spawnSync`, `ret.error` can only be either an `Error` or `undefined`, so it it fails this would always be "instead of an error there is undefined", which does not seem very helpful... I think changing the message to `child process does not error when maxBuffer is set to 1` would be easier to understand.

chrbergert replied:
Good point, thank you

gireeshpunathil commented:
@joyeecheung - PTAL
CI: https://ci.nodejs.org/job/node-test-pull-request/11263/

gireeshpunathil commented on test/parallel/test-child-process-spawnsync-maxbuf.js:
```diff
@@ -16,7 +16,8 @@ const args = [
 
-  assert.ok(ret.error, 'maxBuffer should error');
+  assert.ok(ret.error, 'child process does not error \
+when maxBuffer is set to 1');
```
this your last commit introduced non-standard indentations. Please will you correct it? The previous commit was fair enough, dont know why you pushed again

chrbergert replied:
thanks for the review. Sorry, that I have to change it again. I only wanted to fix the error that the linter was throwing, where i exceeded the maximum line length of 80.

gireeshpunathil commented:
Now that you pushed a commit while the CI was running (which would have picked up the older / newer commit depending on the timing of the push), the CI result is going to be inconclusive ping @Trott to confirm if that is the case.

Trott commented:
Hi @chrbergert! Thanks for the PR and welcome. A few things:

* Looking at this file and the task that was given out, I think this probably wasn't a good task. I'm really sorry about that. But let's keep going because there's stuff to do on this file anyway, and so we'll shift gears to that.

* This fails linting because of the space before the close parentheses.

* I'm not sure the new message is significantly better than the old message. I don't think the old message is as problematic as originally believed. Again, this is where I messed up curating tasks, so sorry about that!

So, a few things:

* To check for lint errors, run `make lint-js`. It will take 20 or 30 seconds the first time, but should take only a second or two on subsequent runs because it caches results of unchanged files.

* I'd say let's revert the error message change (or not if you or @joyeecheung think the new message is better--I personally think they're about the same so I'm fine with either one--one thing about the new error message is it )

* If you *do* want to keep the new message, you can avoid the string concatenation like this:

```js
assert.ok(ret.error,
          'child process does not error when maxBuffer is set to 1');
```

I think I'd prefer this:

```js
assert.ok(ret.error, 'error expected when maxBuffer is set to 1');
```

* Whether or not you change the contents of that assertion message, here are some changes to be made in the test, if you don't mind:

  * Remove the comment `// This is actually not os.EOL?`. Doesn't seem to be important/relevant.
  * Add a comment near the top (under `require('common')` explaining what the test does:

```js
// This test checks that the maxBuffer option for child_process.spawnSync()
// works as expected.
```

chrbergert commented:
Hi together,

first of all, thanks for all the input and the warm welcome. 
I now did the changes to the file like suggested:
- removed initial change for error message
- removed comment that wasn't relevant/important
- added comment that explains what the test does

Before I ran lint-js locally and experimented a bit with it. Think I got the process for this, finally.

So, hope it goes through now and thanks again helping me doing my first contribution to this awesome project.

Trott commented:
CI: https://ci.nodejs.org/job/node-test-pull-request/11378/

Trott commented:
CI was good except for Raspberry Pi devices which have (had?) been having build problems. Let's try again: https://ci.nodejs.org/job/node-test-commit-arm-fanned/12541/

Trott commented:
Landed in 21a7459d490d832d5d547833d527083e460999e1. 
Thanks for the contribution! 🎉
